flink-connectors/flink-connector-files/src/main/java/org/apache/flink/connector/file/src/reader/FileRecordFormat.java

```diff
@@ -34,7 +34,14 @@
 import java.io.Serializable;
 
 /**
- * A reader format that reads individual records from a file.
+ * This interface is Deprecated, use {@link StreamFormat} instead. The main motivation for removing
+ * it is the inherent design flaw in the batching of FileRecordFormat: StreamFormat can guarantee
+ * that only a certain amount of memory is being used (unless a single record exceeds that already),
+ * but FileRecordFormat can only batch by the number of records. By removing FileRecordFormat, we
+ * relay the responsibility of implementing the batching to the format developer; they need to use
+ * BulkFormat and find a better way than batch by number of records.
```

Comment:
I am not sure we really need this docstring here. If someone is interested in the deprecation reason one can open the ticket which is usually linked in the commit message.

Reply:
I am open about it and totally fine to remove it. @AHeise WDYT? I thought this case is a little different. Putting this info in Javadoc will give developers the hint while using StreamFormat or BulkFormat as replacement.

Reply:
I like to give more information to the developers. However, either rephrase it to match JavaDoc style or just link the ticket.